components/server/src/bitbucket-server/bitbucket-server-token-validator.ts

```diff
@@ -34,25 +34,18 @@ export class BitbucketServerTokenValidator implements IGitTokenValidator {
             console.error(error);
         }
 
-        if (found) {
-            writeAccessToRepo = false;
-            const username = await this.api.currentUsername(token);
-            const userProfile = await this.api.getUserProfile(token, username);
-            if (owner === userProfile.slug) {
-                writeAccessToRepo = true;
-            } else {
-                const permission = await this.api.getPermission(token, {
-                    repoKind: repoKind as any,
-                    owner,
-                    username,
-                    repoName: repo,
-                });
-                if (permission && ["REPO_WRITE", "REPO_ADMIN", "PROJECT_ADMIN", ""].includes(permission)) {
-                    writeAccessToRepo = true;
-                }
-            }
+        if (!found) {
+            return {
+                found,
+                isPrivateRepo,
+                writeAccessToRepo,
+            };
         }
 
+        // We can't check REPO_WRITE permission on BitBucketServer since there's no open api for it if we don't change data
```

Comment:
Yes we can!

You can use the [Search for repositories](https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-repository/#api-api-latest-repos-get) endpoint like this  /rest/api/1.0/repos?permission=REPO_WRITE&name=foobar-read-only to test for write permissions, if the project+repository name are known.

Please change the comment to make clear, that we're doing a short cut here which is good enough as long as we request all scopes at once. If we would to change the scope elevation for BBS, it's better to fix the `getPermission` implementation, but we can proceed with removing for now. 
<img width="334" alt="Screenshot 2023-08-29 at 08 02 58" src="https://github.com/gitpod-io/gitpod/assets/914497/986ec367-0cc2-432c-83e8-c9b91b5b120f">

Reply:
Changed to use repo search to verify REPO_WRITE permission